src/verifySignedMessage.js

```diff
@@ -0,0 +1,141 @@
+const recoverSigner = require('./recoverSigner')
+const typedDataEIP712 = require('./typedDataEIP712')
+const proxyAccountFromOwner = require('./proxyAccountFromOwner')
+const { toChecksumAddress, isAddress } = require('web3-utils')
+const encodeFunctionCall = require('./encodeFunctionCall')
+
+const metaDelegateCallParamTypes = [
+  { name: 'to', type: 'address' },
+  { name: 'data', type: 'bytes', calldata: true},
+]
+
+const verifySignedMessage = (signedMessage) => {
+  const { accountAddress, signature, message, signer, EIP712TypedData, functionName, signedParams } = signedMessage
+
+  if (!accountAddress) {
+    throw new Error('Account Address not provided in signedMessage')
+  }
+
+  if (!signature) {
+    throw new Error('Signature not provided in signedMessage')
+  }
+
+  if (!message) {
+    throw new Error('Message not provided in signedMessage')
+  }
+
+  if (!signer) {
+    throw new Error('Signer not provided in signedMessage')
+  }
+
+  if (!functionName) {
+    throw new Error(`No functionName specified in signedMessage`)
+  }
+
+  if (functionName !== 'metaDelegateCall') {
+    throw new Error(`Unsupported functionName \'${functionName}\', only \'metaDelegateCall\' supported`)
+  }
+
+  if (!signedParams || signedParams.length === 0 || !signedParams[1]) {
+    throw new Error('signedParams not provided in signedMessage')
+  }
+
+  if (!EIP712TypedData) {
+    throw new Error('EIP712TypedData not provided in signedMessage')
+  }
+
+  const callData = signedParams[1].callData
+  if (!callData) {
+    throw new Error('CallData not provided in signedMessage Params')
+  }
+
+  const recoveredSigner = recoverSigner({ signature, typedDataHash: message})
+  if (recoveredSigner !== signer) {
```

Comment:
these should be wrapped in `toChecksumAddress` to make the check case insensitive .. see the computed address check below